pkg/collector/corechecks/snmp/internal/checkconfig/config_metric.go

```diff
@@ -25,6 +25,8 @@ type SymbolConfig struct {
 	MatchPattern         string `yaml:"match_pattern"`
 	MatchValue           string `yaml:"match_value"`
 	MatchPatternCompiled *regexp.Regexp
+
+	ScaleFactor float64 `yaml:"scale_factor"`
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏I think adding a higher level test can can help coverage and ensuring scale_factor works fine e2e.

For example by testing from a profile by adding scale factor here:
- https://github.com/DataDog/datadog-agent/blob/0d20d3db2732ab6f91d068ff5245a86ae2b16149/pkg/collector/corechecks/snmp/internal/test/conf.d/snmp.d/profiles/f5-big-ip.yaml#L48-L53 (test for scalar metric) 
- https://github.com/DataDog/datadog-agent/blob/0d20d3db2732ab6f91d068ff5245a86ae2b16149/pkg/collector/corechecks/snmp/internal/test/conf.d/snmp.d/profiles/_generic-if.yaml#L47-L48 (test for column metric)

Adding scale_factor to those profiles will require changing few tests files.